### PR TITLE
TF-1432: Robotiq Refactor

### PR DESCRIPTION
**Pull request by adamcalhoon**
The gripper driver exposed low-level implementation details in order for
    the front end to be able to determine the open/close state of the
    gripper.
    
    Rather than expose these details, the gripper now exposes a service that
    the front end can call to query the gripper open/close state.

### CODE REVIEW

File: robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py [hunk 1]
```diff
@@ -122,6 +125,8 @@ def __init__(self):
         self.reset_thread = None
         self.max_closed = 255
         self.max_open = 3
+        self.grip_state_lock_ = Lock()
```
Comment (mechterp10) [inline robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py:128]: For a variety of reasons, I would prefer you do not initialize this service until the gripper is actually connected. So please move it out of the __init__ and into a function that gets called in the gripper node's initialize_comms.

**Review by adamrr — APPROVED** [review]

File: robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py [hunk 2]
```diff
@@ -498,6 +497,44 @@ def send_gripper_command(self, goal, parse=True):
...
+                    resp.state = GetRobotiqGripStateResponse.OPEN
+                elif moved and registers.gOBJ == 2:
+                    # Fingers stopped due to contact while closing
+                    resp.state = GetRobotiqGripStateResponse.CLOSED
+                elif moved and registers.gOBJ == 3:
+                    # Fingers moved to the requested position without detecting an
+                    # object. Use the gripper position to determine the current
+                    # grip state.
+                    position = int(registers.gPO)
+                    position_median = int(abs(self.max_closed - self.max_open) / 2.0)
```
Comment (mechterp10) [inline robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py]: I would add back in the calibration offsets here as that will shift the median.

**Review by mechterp10 — CHANGES_REQUESTED** [review]

File: robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py (hunk 2 above)
Comment (adamcalhoon) [reply robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py]: Just so I'm clear, you mean:
`position_median = int(abs(self.max_closed - self.max_open) / 2.0) + self.max_open`

Comment (adamcalhoon) [reply robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py]: or did you mean:
`position_median = int(abs(self.max_closed - self.max_open) / 2.0) + (255 - (self.max_closed - self.max_open)`

Comment (mechterp10) [reply robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py]: Sorry. I confused myself. What you had originally is fine. I forgot that max_open and max_closed are updated on calibration. That confusion is probably due to a poor naming choice on my part when it was initially written.

Comment (adamcalhoon) [reply robotiq_c_model_control/src/robotiq_c_model_control/robotiq_gripper_action_interface.py]: OK! I was scratching my head trying to come up with something :)

**Review by mechterp10 — APPROVED** [review]

**Review by michelle-avery — APPROVED** [review]

**Review by sje-rr — APPROVED** [review]

**Pull request merged**